packages/devtools_app/lib/src/debugger/flutter/debugger_controller.dart

```diff
@@ -126,6 +135,34 @@ class DebuggerController extends DisposableController
     commonScriptPrefix = scriptPrefix;
   }
 
+  /// Append to the stdout / stderr buffer.
+  void appendStdio(String text) {
+    const int kMaxLogItemsLowerBound = 5000;
+    const int kMaxLogItemsUpperBound = 5500;
+
+    // Parse out the new lines and append to the end of the existing lines.
+    var lines = _stdio.value.toList();
+    final newLines = text.split('\n');
+
+    if (lines.isNotEmpty && !lines.last.endsWith('\n')) {
+      lines[lines.length - 1] = '${lines[lines.length - 1]}${newLines.first}';
+      if (newLines.length > 1) {
+        lines.addAll(newLines.sublist(1));
+      }
```

Comment:
Why is this needed? This is causing errors in my console to be fused to the last line of the log:

```
Error: Could not launch tel:
...
    at http://localhost:34077/dart_sdk.js:39165:9Error: Could not launch tel:
...                                             ^^ Here :(
    at http://localhost:34077/dart_sdk.js:39165:9

```

Reply:
The stdio that the lines represent may not end in a newline. If not, we just append the new stdio chars to the end of the last line.

There could very well be issues w/ the logic here. Basically what we're trying to do is stream chars into this console, and only insert newlines if they're  part of the incoming stdio.

Reply:
I couldn't find any cases where this was true, but I guess maybe with colored input it might change? (In my case this is making newer logs harder to find / I thought I had broken something :P)